### PR TITLE
CFE-3573: Assertion error no longer logs empty error msg

### PR DESCRIPTION
**Pull request by larsewi**
Promise module python library now handles all exception classes and additionally logs the exception type followed by the message. This way we at least get a hint in case of an assertion error without any message.

### CODE REVIEW

**Review by vpodzime — APPROVED** [review]
Looks good to me.

File: docs/custom_promise_types/cfengine.py [hunk 1]
```diff
@@ -137,6 +137,9 @@ def _handle_validate(self, promiser, attributes):
         except ValidationError as e:
             self.log_error(e)
             self._result = Result.INVALID
+        except Exception as e:
+            self.log_error(f"{type(e).__name__}: {e}")
+            self._result = Result.INVALID
```
Comment (olehermanse) [inline docs/custom_promise_types/cfengine.py]: According to the spec, I think this should be log level critical, and the result should be error:

https://github.com/cfengine/core/blob/master/docs/custom_promise_types/modules.md#results

**Review by olehermanse — APPROVED** [review]
Makes sense, one comment about following the spec, but not super critical.

File: docs/custom_promise_types/cfengine.py (hunk 1 above)
Comment (larsewi) [reply docs/custom_promise_types/cfengine.py]: > Makes sense, one comment about following the spec, but not super critical.
>> According to the spec, I think this should be log level critical, and the result should be error:

It's not super critical, but it should be critical :laughing: \
I will fix !

**Review by olehermanse — APPROVED** [review]
LGTM, but should not have a Changelog entry - doesn't change anything in the released packages.

**Review by olehermanse — APPROVED** [review]

**Pull request merged**